presto-mysql/src/main/java/com/facebook/presto/plugin/mysql/MySqlClientModule.java

```diff
@@ -16,19 +16,38 @@
 import com.facebook.presto.plugin.jdbc.BaseJdbcConfig;
 import com.facebook.presto.plugin.jdbc.JdbcClient;
 import com.google.inject.Binder;
-import com.google.inject.Module;
 import com.google.inject.Scopes;
+import com.mysql.jdbc.Driver;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
 
+import java.sql.SQLException;
+import java.util.Properties;
+
+import static com.google.common.base.Preconditions.checkState;
 import static io.airlift.configuration.ConfigBinder.configBinder;
 
 public class MySqlClientModule
-        implements Module
+        extends AbstractConfigurationAwareModule
 {
     @Override
-    public void configure(Binder binder)
+    protected void setup(Binder binder)
     {
         binder.bind(JdbcClient.class).to(MySqlClient.class).in(Scopes.SINGLETON);
-        configBinder(binder).bindConfig(BaseJdbcConfig.class);
+        BaseJdbcConfig config = buildConfigObject(BaseJdbcConfig.class);
+        ensureCatalogIsEmpty(config.getConnectionUrl());
         configBinder(binder).bindConfig(MySqlConfig.class);
     }
+
+    private static void ensureCatalogIsEmpty(String connectionUrl)
+    {
+        try {
+            Driver driver = new Driver();
+            Properties p = driver.parseURL(connectionUrl, null);
+            checkState(p != null, "Incorrect JDBC URI for MySql connector");
```

Comment:
Change the error message to `"Invalid JDBC URL for MySQL connector"`

(use canonical capitalization for "MySQL" and say "URL" since that's what it's called by JDBC)